server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueFactory.java

```diff
@@ -19,28 +19,33 @@
 
 package org.apache.james.queue.rabbitmq;
 
-import java.io.IOException;
 import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 
 import javax.inject.Inject;
 
 import org.apache.james.queue.api.MailQueueFactory;
 
-import com.github.steveash.guavate.Guavate;
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
 
 public class RabbitMQMailQueueFactory implements MailQueueFactory<RabbitMQMailQueue> {
     private final RabbitClient rabbitClient;
     private final RabbitMQManagementApi mqManagementApi;
     private final RabbitMQMailQueue.Factory mailQueueFactory;
 
+    private final ConcurrentHashMap<MailQueueName, RabbitMQMailQueue> mailQueueStoreLocal;
```

Comment:
There's no way I will understand what this Map goal is when I'll read this class in some months.
The code doesn't explain why it's required so I may very well come and delete it to make the class stateless.

Reply:
Before that, everytime we call `rabbitMQMailQueueFactory.createQueue("SPOOL")`. It create a new `RabbitMQMailQueue` instance, that leads to the duplication multiple register `gauge size` inside the `RabbitMQMailQueue` constructor. To prevent that duplication, I created a local `ConcurrentHashMap<MailQueueName, RabbitMQMailQueue>` like a cache. Each time `rabbitMQMailQueueFactory.createQueue("SPOOL")`, instance from the cache will be returned instead of creating a new one :)

Reply:
Thank you for the explanation, now we should try to make the code speaks by itself.
I will try this afternoon if I find enough time.